### ISSUE TITLE
Cannot create new filament: You have not selected a printer or preset yet. Please select at least one.

### ISSUE THREAD

**Issue by ThepHuck** [SoftFever/OrcaSlicer#6719]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

2.1.1

### Operating System (OS)

Windows

### OS Version

Windows 10 Pro Version	22H2 build 19045.4842

### Additional system information

Intel Core i7-4790k
16GB of RAM
GeForce GTX 1060 6GB
SSD hard drive

### Printer

Elegoo Neptune 4 Max running OpenNept4une

### How to reproduce

Click Filament drop down
Click Add/Remove filament
Click Custom Filaments tab
Click Create New
Provide Basic Information
Selected Create based on current filament (Same behavior if selecting "Copy Current Filament Preset")
Select existing filament
click create
![create-filament](https://github.com/user-attachments/assets/e23438ed-a171-463b-83ed-e2d390185cae)
![create-filament-error](https://github.com/user-attachments/assets/e8dbb57c-eb95-4b9a-a313-7a9f00aaafa2)

### Actual results

error "You have not selected a printer or preset yet. Please select at least one." with only a "yes" button.

### Expected results

Create new filament preset.

### Project file & Debug log uploads

[debug_Mon_Sep_09_14_33_46_1512.log.zip](https://github.com/user-attachments/files/16948803/debug_Mon_Sep_09_14_33
[Nozzle temperature test.zip](https://github.com/user-attachments/files/16948825/Nozzle.temperature.test.zip)
_46_1512.log.zip)

### Checklist of files to include

- [X] Log file
- [X] Project file

**Comment by tlhintoq** [comment]
Duplicate.  There's half a dozen tickets for "can't create new filament"
I opened this one back in 2023
https://github.com/SoftFever/OrcaSlicer/issues/3242

**Comment by markaudacity** [comment]
Still having this problem in October 2024.

**Comment by KwisScram** [comment]
Me also in november 2024!!!

**Comment by mdinnocenzo** [comment]
Currently on release 2.2.0 and I believe that it wants a printer to be selected.  I have noticed that when you try custom filaments it does not appear that the window is resizing to allow for complete selection of the options that pertain to this operation.

**Comment by maindrive** [comment]
I'm having the same issue, I cant believe this has not been resolved yet, this is a big issue for me.

**Comment by blockx** [comment]
Um, is it really still impossible to create a new filament in Orca 2.2.0 in 2025 ?  Is this receiving *any* attention?

**Comment by WeevilKnight** [comment]
Same issue here. I'm on the latest, 2.2.0. Is anyone at support looking at this, or, are they stumped?

**Comment by IsNotMyUserName** [comment]
It seems you are missing something here...
![Image](https://github.com/user-attachments/assets/8ac9647a-7597-47af-95ac-ad00252af397)

I get the same result "you have not selected a printer or preset yet..." When I try to export printer settings or configs from the "Preset" list

When I made a copy it ends up in "User presets" then there was no issue export stuff...

![Image](https://github.com/user-attachments/assets/0d233301-90d8-4f4c-acfc-ba3d2b7225f5)


So look out for that hidden feature that Defaults are no good (branded shits) but your own profile is better.


Same goes for filaments, your own copies ends up in "user presets" vs "system presets"


Not very clear, but we are learning...

**Comment by blockx** [comment]
> It seems you are missing something here... ![Image](https://github.com/user-attachments/assets/8ac9647a-7597-47af-95ac-ad00252af397)
> 
> I get the same result "you have not selected a printer or preset yet..." When I try to export printer settings or configs from the "Preset" list
> 
> When I made a copy it ends up in "User presets" then there was no issue export stuff...
> 
> ![Image](https://github.com/user-attachments/assets/0d233301-90d8-4f4c-acfc-ba3d2b7225f5)
> 
> So look out for that hidden feature that Defaults are no good (branded shits) but your own profile is better.
> 
> Same goes for filaments, your own copies ends up in "user presets" vs "system presets"
> 
> Not very clear, but we are learning...

That response seems wholly impertinent ?

**Comment by mikebryl** [comment]
You have to select the filament in "System Filaments" first. Now it is possible to create a custom filament profile.

![Image](https://github.com/user-attachments/assets/6306b157-8388-4798-8e5a-c18a735376c3)

**Comment by steadmanc** [comment]
I recreated the above post.  I added the Generic Klipper Printer as a Custom Printer.  Then when I add a custom filament, it shows that generic printer as the only selectable option.  It does not show the default printers in that list.

**Comment by gelse** [comment]
I have the same problem. Using KDE/wayland on a monitor with scaling. Rest of the info is in the screenshot.

![Image](https://github.com/user-attachments/assets/009bee07-0cbc-43c4-8bc9-0ec04e8dac51)

**Comment by jpfortis** [comment]
2.3.0 Mac OS.. "You have not selected a printer or preset yet. Please select at least one." nonsense

**Comment by Karl-Barx** [comment]
Dude, how old is this issue now? Crazy, maybe its time to get back on to prusa slicer idk

**Comment by blockx** [comment]
> Dude, how old is this issue now? Crazy, maybe its time to get back on to prusa slicer idk

Yes, the responses here denying the existence of the problem really don't lend confidence in Orca development.  Before there is any chance an issue can be resolved it is necessary to acknowledge the issue exists...

**Comment by cmyk** [comment]
What the heck? 2.3.1 on MacOS and this feature blocks me from adding custom filaments.
Seems the dialog box is cut off and has no scrolling.

![Image](https://github.com/user-attachments/assets/8c506afd-61e7-4704-8834-28fbb9d57ba0)

**Comment by blockx** [comment]
> What the heck? 2.3.1 on MacOS and this feature blocks me from adding custom filaments. Seems the dialog box is cut off and has no scrolling.
> 
> ![Image](https://github.com/user-attachments/assets/8c506afd-61e7-4704-8834-28fbb9d57ba0)

Yep.  It's f'ed, right enough...

**Comment by cmyk** [comment]
> > What the heck? 2.3.1 on MacOS and this feature blocks me from adding custom filaments. Seems the dialog box is cut off and has no scrolling.
> > ![Image](https://github.com/user-attachments/assets/8c506afd-61e7-4704-8834-28fbb9d57ba0)
> 
> Yep. It's f'ed, right enough...

Most likely an easy fix. Lemme see....

**Comment by ArrowThief** [comment]
I found a workaround for anyone still having this issue.
 
 1. Add a Generic filament.

<img width="240" height="218" alt="Image" src="https://github.com/user-attachments/assets/90c91df4-86d6-443e-b7bf-91ff3175c7aa" />

2. Select the filament and then edit it. 

<img width="573" height="90" alt="Image" src="https://github.com/user-attachments/assets/e88baf78-c5c2-478c-92ad-49f118189556" />

3. change settings as needed and save it. 

<img width="620" height="60" alt="Image" src="https://github.com/user-attachments/assets/ed401dca-c7f2-4c63-a8ab-e5aaa4eb6b4f" />

You should now have a custom filament you can change as needed. Don't forget to set compatible machines under the "Dependencies" tab. Otherwise the filament will only be available under the printer you had selected when creating it. 
Hope this helps someone.

**Comment by armaguedes** [comment]
I've only ever used v2.3.0 on Windows 11, and I've always had this problem; the bottom of the "Create Filament" is never visible, and I have to resort to adding a generic filament, and then editing its parameters.

**Comment by blockx** [comment]
> I've only ever used v2.3.0 on Windows 11, and I've always had this problem; the bottom of the "Create Filament" is never visible, and I have to resort to adding a generic filament, and then editing its parameters.

Yes, thanks Armaguedes, the problem does definitely exist.  I'm sure I speak for all users when I say that we are grateful for the work the developers and maintainers do in creating this great software for our hobby.   I understand that there is a need to prioritise and hope this will soon make it to the top of the tree in terms of getting fixed, it is very visible and does come up quite a bit.

Thanks again,

**Comment by jbalius** [comment]
Still same issue in 2025 July with latest OrcaSlicer

<img width="1177" height="785" alt="Image" src="https://github.com/user-attachments/assets/66e51aca-9111-466e-ae39-2fa263a0c389" />

**Comment by jbalius** [comment]
IMHO workaround solution, as this bug is unlikely to be fixed (bug is almost 2 years old already with no progress):

* In filaments select generic filament of same type
* Click Edit button on the right of selection
* In popup top right next to the same generic filament dropdown click save
* Popup will ask for a new name
* Type in your new name and save

The custom type filament will appear in you filament dropdown box. You can further edit it from there and set custom properties to it. I think this should reproduce same outcome as that buggy filament preset creation.

**Comment by blockx** [comment]
Just to thank all the relevant devs for the attention this received, looking forward to trying the fix in the next release...

**Comment by nocturneop15** [comment]
> Just to thank all the relevant devs for the attention this received, looking forward to trying the fix in the next release...

There was no attention on this issue and it was NOT fixed by so marked commit, it resides still in the latest nightly. 2.3.2 .
 It is being ignored for years. In fact, the filament management design is complete mess in Orca, it really sucks and needs a total overhaul - it is strange as the rest of this slicer is magnificent.

**Comment by MilitantTurtle** [comment]
Just came across this issue on latest release 2.3.2 stable, the filament manager in general seems to be broken, can't add a system preset or a custom preset.

**Comment by It5Daniel** [comment]
I found the solution:

1. Open your **current filament settings**.
2. Make sure the selected **Type** is exactly the same as the type you want to use later in _**Custom Filament > Create Filament**_.
3. Click Save Preset, **BEFORE** clicking OK, **enable Detach from parent**.
5. Go back to **Custom Filament > Create Filament**.
6. Select the **exact same Type** that you selected in the filament settings.

and VOLLÀ, you should now be able to select the preset under **Add Filament Preset under this filament** without getting the error.
